extensions-contrib/opentelemetry-extensions/src/main/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReader.java

```diff
@@ -0,0 +1,222 @@
+                .stream()
+                .map(OpenTelemetryMetricsProtobufReader::intDataPointToNumberDataPoint)
+                .map(dataPoint -> parseNumberDataPoint(dataPoint, metric.getName(), resourceAttributes))
+                .collect(Collectors.toList());
+      }
+      case SUM: {
+        return metric.getSum()
+                .getDataPointsList()
+                .stream()
+                .map(dataPoint -> parseNumberDataPoint(dataPoint, metric.getName(), resourceAttributes))
+                .collect(Collectors.toList());
+      }
+      case GAUGE: {
+        return metric.getGauge()
+                .getDataPointsList()
+                .stream()
+                .map(dataPoint -> parseNumberDataPoint(dataPoint, metric.getName(), resourceAttributes))
+                .collect(Collectors.toList());
+      }
+      // TODO Support HISTOGRAM and SUMMARY metrics
+      default:
+        throw new IllegalStateException("Unexpected value: " + metric.getDataCase());
+    }
+  }
+
+  private static NumberDataPoint intDataPointToNumberDataPoint(IntDataPoint dataPoint)
+  {
+    return NumberDataPoint.newBuilder()
+            .setTimeUnixNano(dataPoint.getTimeUnixNano())
+            .setAsInt(dataPoint.getValue())
+            .addAllAttributes(dataPoint.getLabelsList()
+                    .stream()
+                    .map(label -> KeyValue.newBuilder()
+                            .setKey(label.getKey())
+                            .setValue(AnyValue.newBuilder().setStringValue(label.getValue()))
+                            .build())
+                    .collect(Collectors.toList()))
+            .build();
+  }
+
+  private InputRow parseNumberDataPoint(NumberDataPoint dataPoint,
+                                        String metricName,
+                                        List<KeyValue> resourceAttributes)
+  {
+    long timeUnixMilli = TimeUnit.NANOSECONDS.toMillis(dataPoint.getTimeUnixNano());
+    List<KeyValue> metricAttributes = labelsToAttributes(dataPoint);
+    Number value = getValue(dataPoint);
+
+    int capacity = resourceAttributes.size()
+            + metricAttributes.size()
+            + 2; // metric name + value columns
+
+    Map<String, Object> event = Maps.newHashMapWithExpectedSize(capacity);
+    event.put(this.metricDimension, metricName);
+    event.put(this.valueDimension, value);
+
+    resourceAttributes.stream()
+            .filter(ra -> this.attributeMembership.test(this.resourceAttributePrefix + ra.getKey()))
+            .forEach(ra -> event.put(this.resourceAttributePrefix + ra.getKey(), ra.getValue().getStringValue()));
+
+    metricAttributes.stream()
+            .filter(ma -> this.attributeMembership.test(this.metricAttributePrefix + ma.getKey()))
+            .forEach(ma -> event.put(this.metricAttributePrefix + ma.getKey(), ma.getValue().getStringValue()));
+
+    List<String> dimensions = this.schemaDimensions.isEmpty() ? new ArrayList<>(event.keySet()) : this.schemaDimensions;
```

Comment:
it's not enough to take the event's keySet, we also need to apply dimension exclusions from the dimension spec to exclude any dimensions the user wants to exclude. Again, see the opencensus format for how that's done